Enable lvgl example for g1120b0mipi rt700

KATE-WANG-NXP opened the pull request:
1.Fix bug in drivers: mipi_dsi: dsi_mcux_2l when using NXP DCNano DBI driver, previously the pitch>width situation is not supported properly.
2.Enable lvgl example for g1120b0mipi on mimxrt798s

danieldegrasse commented on drivers/display/display_rm67162.c:
```diff
@@ -450,20 +450,28 @@ static int rm67162_write(const struct device *dev, const uint16_t x,
 	src = buf;
 	first_cmd = true;
 
+#ifdef CONFIG_MIPI_DSI_MCUX_NXP_DCNANO_LCDIF
```
I'd really prefer to not have the bus details leak into the driver like this. Is there a way we can get LVGL to allocate buffers that are contiguous to avoid this issue coming up?

dleach02 replied:
@danieldegrasse, instead of pushing the responsibility of buffer allocation structure to LVGL, shouldn't we have some awareness of the underlying LCDIF capabilities so that this driver would be responsible for handling any buffer list structure? 

Seems like something that could be handled as a property of the device interface but it isn't clear to me how the layering here gets handled.

danieldegrasse replied:
Yeah that would be ideal here, just might be harder to add :) @KATE-WANG-NXP how does the RT700 DCNANO actually handle the non-contiguous buffer here? I'm not seeing anywhere in the code where we pass the pitch of the buffer to the LCDIF driver, so I'm not sure how the driver would know when to start a newline while writing to the display.

KATE-WANG-NXP replied:
You are right the pitch is not passed to the low level driver properly. After more tests with the lvgl example with more configurations it turns out my current approach is not feasible to support pitch>width situation. I'm thinking of david's proposal and trying to let the dsi_mcux_2l driver to deal with the non-contiguous buffer in a way so that when DCNano is or is not used the display_rm67162 is not affacted. Will comment to let you know when the update is done, sorry for the trouble for now.

danieldegrasse replied:
Looking at the change it does make sense, but I'm concerned about the number of generic code changes- particularly to LVGL. Honestly in many ways the DCNANO seems to act like a display device (not a MIPI controller) when it is sending display data. Would it make sense to initialize the G1120B0MIPI display using the rm67162 driver, than use a second DCNANO display driver for writing to the display? Similar to what we do for the video mode case for MIPI displays

KATE-WANG-NXP committed:
include: drivers: mipi_dsi: add member in mipi_dsi_msg structure

Add member in mipi_dsi_msg structure to store user data, incase the high
level driver or application needs to pass down extra configuration.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
drivers: display: display_rm67162: update method to transfer frame data

display_rm67162 uses mipi_dsi API(s) to update transfer frame data, some
low level IP can while some cannot handle non-contiguous buffer(the frame
pitch is larger than frame width). In this case, pass the frame descriptor
as the user data in the message to the low level driver and let it
handle how many data to send. The display_rm67162 driver only need to
set the address correctly for each transfer according to the pitch/width.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
drivers: mipi_dsi: dsi_mcux_2l: support non-contiguous frame buffer

The mipi_dbi_nxp_dcnano_lcdif driver support non-contiguous frame
buffer, so when mipi_dbi_nxp_dcnano_lcdif cannot be used to send
the frame buffer data, the driver needs to check whether the data
is non-contiguous first. If it is then extra methods need to be
applied. First the data of each write must not exceed the line
width, second if each line's data exceeds the max payload, it also
needs to be sent seperately.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
modules: lvgl: support the frame buffer address and stride alignment

Add support for the frame buffer address and stride alignment.
If user set the CONFIG_LV_DRAW_BUF_ALIGN to not be 4, the buffers
shall be allocated aoocrdingly.
If user set the CONFIG_LV_DRAW_BUF_STRIDE_ALIGN to not be 1, the
methods of flushing the display shall set the pitch accordingly.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
samples: modules: lvgl: Enable example for g1120b0mipi on mimxrt798s

Enable lvgl example for g1120b0mipi shield on mimxrt798s board.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
boards: shields: update g1120b0mipi overlay

Add configuration to support touch function in lvgl
examples.

signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP commented:
Rebased main branch.
